gym/spaces/multi_discrete.py

```diff
@@ -63,8 +63,22 @@ def shape(self) -> Tuple[int, ...]:
         """Has stricter type than :class:`gym.Space` - never None."""
         return self._shape  # type: ignore
 
-    def sample(self) -> np.ndarray:
+    def sample(self, mask: np.ndarray = None) -> np.ndarray:
         """Generates a single random sample this space."""
+        if mask is not None:
+            assert isinstance(mask, np.ndarray)
+            assert mask.dtype == np.int8
+            assert mask.shape == self.shape
+
+            multi_mask = [np.where(row) for row in mask]
```

Comment:
This is a good first-pass implementation so I think it's fine here, but do note that its performance might be throttled when we have a large MultiDiscrete action space. See https://github.com/vwxyzjn/gym-microrts/pull/71/files

Reply:
I was thinking about that but I couldn't work out a way of doing it efficiently
You can't use `choice` as you want it for each row individually